addons/binding/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/command/SatelCommandBase.java

```diff
@@ -0,0 +1,191 @@
+        if (response == null) {
+            return false;
+        }
+        if (response.getCommand() == COMMAND_RESULT_CODE) {
+            if (!hasCommandSucceeded(response)) {
+                return false;
+            }
+        } else if (!isResponseValid(response)) {
+            return false;
+        }
+        this.response = response;
+        return true;
+    }
+
+    public void ignoreResponseError() {
+        this.logResponseError = false;
+    }
+
+    /**
+     * Checks whether given response is valid for the command.
+     *
+     * @param response message to check
+     * @return <code>true</code> if given message is valid response for the command
+     */
+    protected abstract boolean isResponseValid(SatelMessage response);
+
+    protected static int bcdToInt(byte[] bytes, int offset, int size) {
+        int result = 0, digit;
+        int byteIdx = offset;
+        int digits = size * 2;
+        for (int i = 0; i < digits; ++i) {
+            if (i % 2 == 0) {
+                digit = (bytes[byteIdx] >> 4) & 0x0f;
+            } else {
+                digit = (bytes[byteIdx]) & 0x0f;
+                byteIdx += 1;
+            }
+            result = result * 10 + digit;
+        }
+        return result;
+    }
+
+    protected boolean hasCommandSucceeded(SatelMessage response) {
+        // validate response message
+        if (response.getCommand() != COMMAND_RESULT_CODE) {
+            logger.error("Invalid response code: {}. {}", response.getCommand(), getRequest());
+            return false;
+        }
+        if (response.getPayload().length != 1) {
+            logger.error("Invalid payload length: {}. {}", response.getPayload().length, getRequest());
+            return false;
+        }
+
+        // check result code
+        byte responseCode = response.getPayload()[0];
+        String errorMsg;
+
+        switch (responseCode) {
+            case 0:
+                // success
+                return true;
+            case 0x01:
+                errorMsg = "Requesting user code not found";
+                break;
+            case 0x02:
+                errorMsg = "No access";
+                break;
+            case 0x03:
+                errorMsg = "Selected user does not exist";
+                break;
+            case 0x04:
+                errorMsg = "Selected user already exists";
+                break;
+            case 0x05:
+                errorMsg = "Wrong code or code already exists";
+                break;
+            case 0x06:
+                errorMsg = "Telephone code already exists";
+                break;
+            case 0x07:
+                errorMsg = "Changed code is the same";
+                break;
+            case 0x08:
+                errorMsg = "Other error";
+                break;
+            case 0x11:
+                errorMsg = "Can not arm, but can use force arm";
+                break;
+            case 0x12:
```

Comment:
Maybe it could make sense to store these in a map instead and look them up from there. You would only need a separate case for success and for codes not being in the map.

Reply:
I am not sure it will be better, now it is in one place, so you can easily check what message is assigned to specific error code. But if you insist, I will change it.